CHANGES.md

```diff
@@ -1,6 +1,429 @@
 Release Changes List
 ====================
 
+TODO:
+* check for docs/master
+* any last-minute contributions / contributors?
+* '' vs. ``
+* check users guide code
+* remove empty sections
+
+version 1.21.0
+==============
+
+Twenty-fourth public release of Chapel, April 9, 2020
+
+Highlights (see subsequent sections for further details)
+--------------------------------------------------------
+
+Syntactic/Naming Changes
+------------------------
+* `private config`s must now be fully qualified when set on the command-line
+* made minor adjustments to the argument names of `string` and `bytes` methods
+
+Semantic Changes / Changes to Chapel Language
+---------------------------------------------
+* made `use` private by default
+  (see https://chapel-lang.org/docs/1.21/language/spec/statements.html#the-use-statement)
+* sub-modules no longer have lexical visibility into their parent modules
+  (see https://chapel-lang.org/docs/1.21/language/spec/modules.html#nested-modules)
+* the `locale` type now has value semantics and a default value of `Locales[0]`
+* values of type `string` are now UTF8-validated
+* `bytes.this` and `bytes.these` now return/yield `unit(8)` instead of `bytes`
+  (see https://chapel-lang.org/docs/1.21/builtins/Bytes.html)
+* stopped generating `init=` functions for records with a user-defined `init=`
+* an `init=` containing compilerError() now indicates the type is not copyable
+  (TODO: PR #14887 - not sure where this should be documented?)
+* added support for split initialization of variables, refs, types, and params
+  (see https://chapel-lang.org/docs/1.21/language/spec/variables.html#split-initialization)
+  (TODO language evolution link)
+* added support for copy elision transformations
+  (TODO: where to document?)
+* improved `out` intents to support split initialization and to reduce copies
+  (TODO language evolution link? spec?)
+* `record` temporaries may now be deinitialized at the end of a statement
+  (TODO: where to document?)
+* compile-time (`type`, `param`) methods now require `overload` when overloaded
+* made `enum` casts that may fail throw an error rather than halt
+  (see https://chapel-lang.org/docs/1.21/language/spec/conversions.html#explicit-enumeration-conversions)
+* added an execution-time check to guard against resizing arrays of non-nilable
+  (see https://chapel-lang.org/docs/1.21/language/spec/classes.html#class-values)
+* made the compiler no longer tolerate assignment overloads for classes
+
+New Features
+------------
+* added support for `import` statements as a more precise way of using modules
+  (see <TODO> doc link)
+* added support for renaming a module in its `use` statement
+  (see <TODO> doc link)
+* added explicit relative `import` and `use` chains via `this.` and `super.`
+  (TODO: where to document?)
```

Comment:
https://chapel-lang.org/docs/master/language/spec/statements.html#the-use-statement